test/admin.service.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { UsersModule } from '../src/modules/users/users.module';
+import { UsersRepository } from '../src/modules/users/users.repository';
+import { AdminsService } from '../src/modules/users/admin.service';
+
+describe('AdminsServise', () => {
+  let adminsServise: AdminsService;
+
+  interface IAdmin {
+    name: string;
+    surname: string;
+    email: string;
+    phone: number;
+    password: string;
+    role: 1;
+  }
+
+  interface IFilterParamsAdmin {
+    email: string | null;
+    name: string | null;
+    surname: string | null;
+    role: 1;
+  }
+
+  interface ISortParams {
+    sortBy: string | null;
+    orderBy: string | null;
+  }
+
+  const MockUsersRepository = {
+    readAllUsers: jest.fn(
+      (IFilterParamsAdmin, ISortParams, limit: number, skip: number) => {
+        return Promise.resolve(Array<IAdmin>());
+      },
+    ),
+  };
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [UsersModule],
+      providers: [
+        AdminsService,
+        {
+          provide: 'SUBSCRIBERS_SERVICE',
+          useValue: {},
+        },
+      ],
+    })
+      .overrideProvider(UsersRepository)
```

Comment:
you don't need this
just add UsersRepository to providers array as `
{           
    provide: getRepositoryToken(UsersRepository)  useValue: MockUsersRepository,}`